src/components/MainContent/MainContent.js

```diff
@@ -43,14 +32,18 @@ function MainContent() {
   }
 
   const chooseResult = () => {
-    if (!data.gotResponse) {
+    if (data.loadingStatus === LOADING) {
       return <Loader />
-    } else if (data.data.length > 0) {
+    } else if (data.loadingStatus === LOADED) {
```

Comment:
no need for `else` se hai sempre un `return` value dentro ogni if (bastano 2 if e puoi rimuovere l'else)